backend_new/src/services/application-flagged-set.service.ts

```diff
@@ -0,0 +1,951 @@
+    const rawAfs = await this.prisma.applicationFlaggedSet.findUnique({
+      where: {
+        id: afsId,
+      },
+      include: {
+        applications: {
+          include: {
+            applicant: true,
+          },
+        },
+        listings: true,
+      },
+    });
+
+    if (!rawAfs) {
+      throw new NotFoundException(
+        `applicationFlaggedSetId ${afsId} was requested but not found`,
+      );
+    }
+
+    return mapTo(ApplicationFlaggedSet, rawAfs);
+  }
+
+  /**
+    this resets the showConfirmationAlert on the application flagged set
+  */
+  async resetConfirmationAlert(afsId: string): Promise<SuccessDTO> {
+    await this.findOrThrow(afsId);
+    await this.prisma.applicationFlaggedSet.update({
+      data: {
+        showConfirmationAlert: false,
+      },
+      where: {
+        id: afsId,
+      },
+    });
+    return {
+      success: true,
+    };
+  }
+
+  /**
+    this will return meta info for a set of application flagged sets
+  */
+  async meta(params: AfsQueryParams): Promise<AfsMeta> {
+    const [
+      totalCount,
+      totalResolvedCount,
+      totalPendingCount,
+      totalNamePendingCount,
+      totalEmailPendingCount,
+    ] = await Promise.all([
+      this.metaHelper(params.listingId),
+      this.metaHelper(params.listingId, FlaggedSetStatusEnum.resolved),
+      this.metaHelper(params.listingId, FlaggedSetStatusEnum.pending),
+      this.metaHelper(
+        params.listingId,
+        FlaggedSetStatusEnum.pending,
+        RuleEnum.nameAndDOB,
+      ),
+      this.metaHelper(
+        params.listingId,
+        FlaggedSetStatusEnum.pending,
+        RuleEnum.email,
+      ),
+    ]);
+
+    return {
+      totalCount,
+      totalResolvedCount,
+      totalPendingCount,
+      totalNamePendingCount,
+      totalEmailPendingCount,
+    };
+  }
+
+  /**
+    helper that builds the meta functions queries
+  */
+  metaHelper(
+    listingId: string,
+    status?: FlaggedSetStatusEnum,
+    rule?: RuleEnum,
+  ): Promise<number> {
+    return this.prisma.applicationFlaggedSet.count({
+      where: {
+        listingId,
+        status,
+        rule,
+      },
+    });
+  }
+
+  /**
+    resolves an application flagged set
+  */
+  async resolve(dto: AfsResolve, user: User): Promise<ApplicationFlaggedSet> {
+    const filter: Prisma.ApplicationFlaggedSetWhereInput[] = [
+      {
+        id: dto.afsId,
+      },
+    ];
+    if (dto.applications?.length) {
+      filter.push({
+        applications: {
+          some: {
+            id: {
+              in: dto.applications.map((application) => application.id),
+            },
+          },
+        },
+      });
+    }
+    const afs = await this.prisma.applicationFlaggedSet.findFirst({
+      where: {
+        AND: filter,
+      },
+      include: {
+        listings: true,
+        applications: {
+          where: {
+            id: {
+              in: dto.applications.map((application) => application.id),
```

Comment:
yeah so its kinda an interesting thing that prisma does
the one in the where clause would only get flagged sets where the application is attached (applications.some.id.i)

the other one (include.applications.where.id.in) is so that when we get the list of application flagged sets back the applications that are returned as part of that call are only those that meet the criteria

without that second one we would just be getting the entire set of applications on the flagged set. Hopefully that makes some sense 

https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#filter-a-list-of-relations

is the documentation for it